src/plugins/intel_cpu/src/compiled_model.cpp

```diff
@@ -201,44 +188,46 @@ ExecNetwork::GraphGuard::Lock ExecNetwork::GetGraph() const {
     return graphLock;
 }
 
-InferenceEngine::IInferRequestInternal::Ptr ExecNetwork::CreateInferRequest() {
-    return CreateAsyncInferRequestFromSync<AsyncInferRequest>();
+std::shared_ptr<ov::ISyncInferRequest> CompiledModel::create_sync_infer_request() const {
+    m_numRequests++;
+    return std::make_shared<SyncInferRequest>(std::static_pointer_cast<const CompiledModel>(shared_from_this()));
 }
 
-std::shared_ptr<ngraph::Function> ExecNetwork::GetExecGraphInfo() {
-    if (_graphs.empty())
-        IE_THROW() << "No graph was found";
+std::shared_ptr<ov::IAsyncInferRequest> CompiledModel::create_infer_request() const {
+    auto internal_request = create_sync_infer_request();
+    auto async_infer_request =
+        std::make_shared<AsyncInferRequest>(std::static_pointer_cast<SyncInferRequest>(internal_request),
+                                            get_task_executor(),
+                                            get_callback_executor());
+    return async_infer_request;
+}
 
-    return GetGraph()._graph.dump();
+std::shared_ptr<const ov::Model> CompiledModel::get_runtime_model() const {
+    if (m_graphs.empty())
+        OPENVINO_THROW("No graph was found");
+
+    return get_graph()._graph.dump();
 }
 
-Parameter ExecNetwork::GetConfigLegacy(const std::string &name) const {
-    if (_graphs.empty())
-        IE_THROW() << "No graph was found";
-    /* legacy implementation return all the parameters which is actually not correct
-     * since they are not reconfigurable. Fixed for new API */
-    Config engConfig = GetGraph()._graph.getConfig();
+ov::Any CompiledModel::get_property(const std::string& name) const {
+    if (m_graphs.empty())
+        OPENVINO_THROW("No graph was found");
+
+    if (name == ov::loaded_from_cache) {
+        return m_loaded_from_cache;
+    }
+
+    Config engConfig = get_graph()._graph.getConfig();
     auto option = engConfig._config.find(name);
     if (option != engConfig._config.end()) {
         return option->second;
-    } else {
-        IE_THROW() << "Unsupported ExecutableNetwork config key: " << name;
     }
-}
 
-/**
- * Only legacy parameters are supported.
- * No RW peroperties supported for new API.
- * All the RO properties are covered with GetMetric() method and
- * GetConfig() is not expected to be called by new API with params from new configuration API.
- */
-Parameter ExecNetwork::GetConfig(const std::string &name) const {
-    /* Internally legacy parameters are used with new API as part of migration procedure.
-     * This fallback can be removed as soon as migration completed */
-    return GetConfigLegacy(name);
+    return get_metric(name);
 }
 
-InferenceEngine::Parameter ExecNetwork::GetMetricLegacy(const std::string &name, const GraphGuard& graph) const {
+ov::Any CompiledModel::get_metric_legacy(const std::string& name, const GraphGuard& graph) const {
+    OPENVINO_SUPPRESS_DEPRECATED_START
```

Comment:
Some metrics like `SUPPORTED_METRICS`, `SUPPORTED_CONFIG_KEYS` and etc. are already handled on the Core side or can be represented by new properties.

Reply:
yes, we will remove such legacy metrics, maybe can put them in https://github.com/openvinotoolkit/openvino/pull/20022